src/test/java/com/wikia/webdriver/testcases/desktop/articlecrudtests/ArticleSourceModeTests.java

```diff
@@ -280,4 +287,71 @@ public void RTE_019_MoreSympolsTools() {
     source.checkSymbolsTools();
     source.submitArticle();
   }
+
+  @Test(groups = {"RTE_draft","draft_saving"})
+  @Execute(asUser = User.SUS_REGULAR_USER3, onWikia = "draftsavetest")
+  public void RTE_draft_save() {
+    String articleName = PageContent.ARTICLE_NAME_PREFIX + DateTime.now().getMillis();
+    SourceEditModePageObject source = new SourceEditModePageObject().openArticle(articleName);
+    source.addContentInSourceMode(articleName);
+//This sleep is necessery for draft to be saved (every 5 sec)
+    try {
+      Thread.sleep(10001);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    }
+    driver.navigate().refresh();
+    AlertHandler.acceptPopupWindow(driver,15);
+
+    source.closeDraftNotification();
+    Assertion.assertStringContains(source.getSourceContent(),articleName);
+  }
+
+  @Test(groups = {"RTE_draft","draft_saving"})
+  @Execute(asUser = User.SUS_REGULAR_USER3, onWikia = "draftsavetest")
+  public void RTE_draft_intervening() {
+    String articleName = PageContent.ARTICLE_NAME_PREFIX + DateTime.now().getMillis();
+    new ArticleContent(User.SUS_REGULAR_USER3).push("An awesome start", articleName);
+    SourceEditModePageObject source = new SourceEditModePageObject().openArticle(articleName);
+    source.addContentInSourceMode(articleName);
+    new ArticleContent(User.SUS_REGULAR_USER).push("Test content", articleName);
+//This sleep is necessery for draft to be saved (every 5 sec)
+    try {
+      Thread.sleep(10001);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    }
+    driver.navigate().refresh();
+    AlertHandler.acceptPopupWindow(driver,15);
+    source.closeDraftNotification();
+    Assertion.assertStringContains(source.getSourceContent(),articleName);
+  }
+
+  @Test(groups = {"RTE_draft","draft_saving"})
+  @Execute(asUser = User.SUS_REGULAR_USER3, onWikia = "draftsavetest")
+  public void RTE_draft_intervening_rejected() {
+    String articleName = PageContent.ARTICLE_NAME_PREFIX + DateTime.now().getMillis();
+
+    new ArticleContent(User.SUS_REGULAR_USER3).push("An awesome start", articleName);
+
+//    String articleName2 = articleName+"D";
+
+    SourceEditModePageObject source = new SourceEditModePageObject().openArticle(articleName);
+    source.addContentInSourceMode(articleName);
+//    source.focusTextArea();
+//    source.addContentInSourceMode(articleName);
+    new ArticleContent(User.SUS_REGULAR_USER).push("Test content", articleName);
+//This sleep is necessery for draft to be saved (every 5 sec)
+
+    try {
+      Thread.sleep(10001);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1148)